lib/RawPreviewBase.php

```
@@ -58,6 +58,10 @@ protected function getThumbnailInternal(File $file, int $maxX, int $maxY): ?IIma
             return null;
         }
 
+        if (exif_imagetype($localPath) === IMAGETYPE_AVIF) {
```

Can you extend getThumbnailInternal. Do the same as with tiff. The extension is enough here I think.
exif_imagetype might not be available (as the exif extension is not a hard requirement of this app).

---

you also need to check that avif is supported in the installed imagick, just like done with tiff:
    private function isTiffCompatible(): bool
    {
        return extension_loaded('imagick') && count(\Imagick::queryformats('TIFF')) > 0;
    }

---

I added `isAvifCompatible()` and also realized I forgot to update `getMimeType()` accordingly.
I could modify `getBestPreviewTag()`, but I think it’s simpler to use
`if (str_ends_with(strtolower($file->getName()), '.avif')) {`
This would eliminate the dependency on `exif_imagetype()` and avoid related issues.

---

You should also explicity check for avif support in imagick. See how I do it with Tiff.

---

Right, just checking `isAvailable()` isn’t enough.
Something like:
```php
if ($this->isAvifCompatible() && str_ends_with(strtolower($file->getName()), '.avif')) {
    // generate Imagick thumbnail
}
```
makes sense.

That would be similar to how `getBestPreviewTag()` handles TIFF:
```php
if ($fileType === 'TIFF' && $this->isTiffCompatible()) {
```
Would you recommend adding an explicit error like this (similar to the TIFF case):
```php
if (!$this->isTiffCompatible()) {
    throw new Exception('Needs imagick to extract TIFF previews');
}
```
or should I silently skip AVIF without throwing an exception?

---

Fail silent I think. If we had good config for this it could throw a warning. But when we are introducing stuff that is not strictly required I think an info logging or nothing is better.

---

Yes, I agree, this case is probably very rare.
If someone is using an outdated Imagick build without AVIF support, it's most likely intentional or a special environment. I only used Imagick here to avoid pulling a large amount of AVIF-handling logic into the app, otherwise I would have stayed with GD as well. Adding a prominent warning or log entry might just create unnecessary noise, so keeping it silent (or only logging at info level) makes sense.

If explicit logging is ever desired, the check could look like this:
```php
if (str_ends_with(strtolower($file->getName()), '.avif')) {
    if ($this->isAvifCompatible()) {
        // generate Imagick thumbnail
    } else {
        $this->logger->warning('Imagick without AVIF support, skipping AVIF preview', ['app' => $this->appName]);
        return null; // or just silently skip
    }
}
```
But for now I’ll leave it silent, as you suggested.

One more question regarding memory cleanup:
In the TIFF code you do not call `$imagick->clear()` or `$imagick->destroy()`.
I added them on the AVIF path just to be safe and avoid leaving large images in memory.
Is that unnecessary because Imagick is destroyed automatically after returning,
or do you recommend explicitly clearing it?

---

No idea if cleanup is needed. As php is request based there is probably no real world benefit to it, but I might be wrong here.